test/e2e/declarative_test_fwk/helper.py

```diff
@@ -110,21 +114,36 @@ def delete(self, reference: k8s.CustomResourceReference) -> None:
         sleep(self.DEFAULT_WAIT_SECS)
         self.wait_for_delete(reference)  # throws exception if wait fails
 
-    def assert_expectations(self, verb: str, input_data: dict, expectations: model.ExpectDict, reference: k8s.CustomResourceReference) -> None:
-        """Asserts custom resource reference inside Kubernetes cluster against the supplied expectations
+    def assert_k8s_resource(self, verb: str, input_data: dict, expectations_k8s: model.ExpectK8SDict, reference: k8s.CustomResourceReference) -> None:
+        """Compare the supplied expectations with custom resource reference inside Kubernetes cluster
 
-        :param verb: expectations after performing the verb (apply, patch, delete)
+        :param verb: compare expectations with k8s resource after performing the verb (apply, patch, delete)
         :param input_data: input data to verb
-        :param expectations: expectations to assert
+        :param expectations_k8s: expectations to assert
         :param reference: custom resource reference
         :return: None
         """
-        self._assert_conditions(expectations, reference, wait=False)
-        # conditions expectations met, now check current resource against expectations
+        self._assert_conditions(expectations_k8s, reference, wait=False)
+        # conditions expectations met, now check current resource against expectations_k8s
         resource = k8s.get_resource(reference)
-        self.assert_items(expectations.get("status"), resource.get("status"))
+        # status and spec assertions with resources from Kubernetes
+        self.assert_items_k8s(expectations_k8s.get("status"), resource.get("status"))
+        self.assert_items_k8s(expectations_k8s.get("spec"), resource.get("spec"))
+
+    def assert_aws_resource(self, verb: str, input_data: dict, expectations_aws: dict, reference: k8s.CustomResourceReference) -> None:
+        """Compare the supplied expectations with aws resource from api calls by boto3
 
-        # self._assert_state(expectations.get("spec"), resource)  # uncomment to support spec assertions
+        :param verb: compare expectations with aws resource after performing the verb (apply, patch, delete)
+        :param input_data: input data to verb
```

Comment:
nit: We aren't using `verb` or `input_data` in this method

Reply:
Deleted them